--- plasma-workspace-5.27.12/kcms/kfontinst/lib/CMakeLists.txt~	2025-01-06 18:57:36.000000000 +0100
+++ plasma-workspace-5.27.12/kcms/kfontinst/lib/CMakeLists.txt	2026-05-26 23:29:29.508289033 +0200
@@ -9,6 +9,7 @@
     KF5::I18n
     KF5::KIOCore
     Fontconfig::Fontconfig
+    Freetype::Freetype
     PRIVATE
     Qt::Xml
     Qt::Gui
--- plasma-workspace-5.27.12/kcms/kfontinst/lib/Fc.h~	2025-01-06 18:57:36.000000000 +0100
+++ plasma-workspace-5.27.12/kcms/kfontinst/lib/Fc.h	2026-05-26 23:30:10.458558962 +0200
@@ -10,6 +10,7 @@
 #include "kfontinst_export.h"
 #include <QUrl>
 #include <fontconfig/fontconfig.h>
+#include <fontconfig/fcfreetype.h>
 
 #if (FC_VERSION < 20200)
 
